Firestore/core/src/firebase/firestore/local/leveldb_remote_document_cache.cc

```diff
@@ -102,13 +146,22 @@ OptionalMaybeDocumentMap LevelDbRemoteDocumentCache::GetAll(
     it->Seek(LevelDbRemoteDocumentKey::Key(key));
     if (!it->Valid() || !current_key.Decode(it->key()) ||
         current_key.document_key() != key) {
-      results = results.insert(key, absl::nullopt);
+      results.Insert(std::make_pair(key, absl::nullopt));
     } else {
-      results = results.insert(key, DecodeMaybeDocument(it->value(), key));
+      const std::string& contents = it->value();
+      tasks.Execute([this, &results, key, contents] {
```

Comment:
Optional: it looks like you can capture `key` by reference? (it seems like `key` is valid as long as `keys`, which is an argument to this function, is valid)